common/src/stack/report-system/command/report/system/tests/test_other_dhcp.py

```diff
@@ -57,7 +57,27 @@ def test_other_dhcp_server(interface):
 	# Set up socket with passed in interface
 	send_dhcp = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
 	send_dhcp.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
-	send_dhcp.setsockopt(socket.SOL_SOCKET, 25, str(interface + '\0').encode('utf-8'))
+
+	# Bail if we can't setup a socket on the current interface
+	# for instance if it's a virtual interface
+	try:
+		send_dhcp.setsockopt(socket.SOL_SOCKET, 25, str(interface + '\0').encode('utf-8'))
+
+	except OSError:
+		pytest.skip(f'Could not setup socket for {interface}')
+
+	mac_addr = pxe_interfaces[interface]
+
+	# If the MAC address isn't in the Stacki database, try elsewhere
+	if not mac_addr:
+		try:
+			mac_file = open(f'/sys/class/net/{interface}/address', 'r')
```

Comment:
Please use a `open` as a context manager here to ensure the file handle is always closed, or use `pathlib.Path(..).read_text()` to get it all in one line.

```
with open(f'/sys/class/net/{interface}/address', 'r') as mac_file:
    mac_addr = mac_file.read().strip()
####
# OR
###
mac_addr = pathlib.Path(f'/sys/class/net/{interface}/address').read_text()
```